evennia/contrib/turnbattle.py

```diff
@@ -0,0 +1,663 @@
+"""
+Simple turn-based combat system
+
+Contrib - Tim Ashley Jenkins 2017
+
+This is a framework for a simple turn-based combat system, similar
+to those used in D&D-style tabletop role playing games. It allows
+any character to start a fight in a room, at which point initiative
+is rolled and a turn order is established. Each participant in combat
+has a limited time to decide their action for that turn (30 seconds by
+default), and combat progresses through the turn order, looping through
+the participants until the fight ends.
+
+Only simple rolls for attacking are implemented here, but this system
+is easily extensible and can be used as the foundation for implementing
+the rules from your turn-based tabletop game of choice or making your
+own battle system.
+
+To install and test, import this module's BattleCharacter object into
+your game's character.py module:
+
+    from evennia.contrib.turnbattle import BattleCharacter
+
+And change your game's character typeclass to inherit from BattleCharacter
+instead of the default:
+
+    class Character(BattleCharacter):
+        
+Next, import the combat commands into your default_cmdsets.py module:
+
+    from evennia.contrib.turnbattle import CmdFight, CmdAttack, CmdRest, CmdPass, CmdDisengage
+
+And add the commands to your default command set:
+
+    #
+    # any commands you add below will overload the default ones.
+    #
+    self.add(CmdFight())
+    self.add(CmdAttack())
+    self.add(CmdRest())
+    self.add(CmdPass())
+    self.add(CmdDisengage())
+
+This module is meant to be heavily expanded on, so you may want to copy it
+to your game's 'world' folder and modify it there rather than importing it
+in your game and using it as-is.
+"""
+
+from random import randint
+from evennia import DefaultCharacter, Command, default_cmds, DefaultScript
+
+"""
+----------------------------------------------------------------------------
+COMBAT FUNCTIONS START HERE
+----------------------------------------------------------------------------
+"""
+def roll_init(character):
+    """
+    Rolls a number between 1-1000 to determine initiative.
+    
+    Args:
+        character (obj): The character to determine initiative for
+        
+    Returns:
+        initiative (int): The character's place in initiative - higher
+        numbers go first.
+        
+    Notes:
+        By default, does not reference the character and simply returns
+        a random integer from 1 to 1000.
+    """
+    return randint(1,1000)
+    """
```

Comment:
docstrings are usually at the top of the function/method/class etc. Probably better to put this info in the Notes section of the docstring instead or, if you want to have it in the  code, make it a normal comment with # markers.